Rungtime error: index out of range [0]

I keep getting this error when I'm trying to create a connection between Digitalstrom and HomeAssisstant.

It only happens when I use the following parameters:
HOME_ASSISTANT_DISCOVERY_ENABLED=true
HOME_ASSISTANT_DISCOVERY_PREFIX=homeassistant

`﻿
panic: runtime error: index out of range [0] with length
goroutine 1 [running]:
github.com/gaetancollaud/digitalstrom-mqtt/digitalstrom_mqtt. (*HomeAssistantMqtt).deviceToHomeAssistantDiscoveryMessage(0x4000011880, {{0x40000b06e8, 0x8}, {0x40000c41e0, 0x18}, {0x40000b6090, 0x22}, {0x36044e, 0x2}, {0x40000b0770, ...},
/home/runner/work/digitalstrom-mqtt/digitalstrom-mqtt/digitalstrom_mqtt/mqtt_homeassistant.go:109 +0xe00
github.com/gaetancollaud/digitalstrom-mqtt/digitalstrom_mqtt.(*HomeAssistantMqtt).publishDiscoveryMessages (0x4000011880)
/home/runner/work/digitalstrom-mqtt/digitalstrom-mqtt/digitalstrom_mqtt/mqtt_homeassistant.go:37 +0x238
github.com/gaetancollaud/digitalstrom-mqtt/digitalstrom_mqtt.(*HomeAssistantMqtt).Start(...)
/home/runner/work/digitalstrom-mqtt/digitalstrom-mqtt/digitalstrom_mqtt/mqtt_homeassistant.go:31
github.com/gaetancollaud/digitalstrom-mqtt/digitalstrom_mqtt.(*DigitalstromMqtt).Start(0x4000065aa0)
/home/runner/work/digitalstrom-mqtt/digitalstrom-mqtt/digitalstrom_mqtt/mqtt.go:95 +0x24c
main.main()
/home/runner/work/digitalstrom-matt/digitalstrom-mqtt/main.go:47 +0x40c`

--- a/digitalstrom_mqtt/mqtt_homeassistant.go
+++ b/digitalstrom_mqtt/mqtt_homeassistant.go
@@ -3,6 +3,7 @@ package digitalstrom_mqtt
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/rs/zerolog/log"
 	"strings"
 
 	"github.com/gaetancollaud/digitalstrom-mqtt/config"
@@ -90,49 +91,57 @@ func (hass *HomeAssistantMqtt) deviceToHomeAssistantDiscoveryMessage(device digi
 	var message map[string]interface{}
 	var topic string
 	if device.DeviceType == digitalstrom.Light {
-		// Setup configuration for a MQTT Cover in Home Assistant:
-		// https://www.home-assistant.io/integrations/light.mqtt/
-		nodeId := "light"
-		topic = hass.discoveryTopic(Light, device.Dsid, nodeId)
-		message = map[string]interface{}{
-			"device": deviceConfig,
-			"name": utils.RemoveRegexp(
-				device.Name,
-				hass.config.RemoveRegexpFromName),
-			"unique_id":         device.Dsid + "_" + nodeId,
-			"availability":      availability,
-			"availability_mode": "all",
-			"command_topic": hass.mqtt.getTopic(
-				"devices",
-				device.Dsid,
-				device.Name,
-				device.OutputChannels[0],
-				"command"),
-			"state_topic": hass.mqtt.getTopic(
-				"devices",
-				device.Dsid,
-				device.Name,
-				device.OutputChannels[0],
-				"state"),
-			"payload_on":  "100.00",
-			"payload_off": "0.00",
-			"qos":         0,
-		}
-		if device.Properties.Dimmable {
-			message["on_command_type"] = "brightness"
-			message["brightness_scale"] = 100
-			message["brightness_state_topic"] = hass.mqtt.getTopic(
-				"devices",
-				device.Dsid,
-				device.Name,
-				device.OutputChannels[0],
-				"state")
-			message["brightness_command_topic"] = hass.mqtt.getTopic(
-				"devices",
-				device.Dsid,
-				device.Name,
-				device.OutputChannels[0],
-				"command")
+		if device.OutputChannels == nil || len(device.OutputChannels) == 0 {
+			log.Warn().
+				Str("dsid", device.Dsid).
+				Str("deviceName", device.Name).
+				Msg("Light device has no output channel")
+		} else {
+			// Setup configuration for a MQTT Cover in Home Assistant:
+			// https://www.home-assistant.io/integrations/light.mqtt/
+			nodeId := "light"
+			topic = hass.discoveryTopic(Light, device.Dsid, nodeId)
+			message = map[string]interface{}{
+				"device": deviceConfig,
+				"name": utils.RemoveRegexp(
+					device.Name,
+					hass.config.RemoveRegexpFromName),
+				"unique_id":         device.Dsid + "_" + nodeId,
+				"availability":      availability,
+				"availability_mode": "all",
+				"command_topic": hass.mqtt.getTopic(
+					"devices",
+					device.Dsid,
+					device.Name,
+					device.OutputChannels[0],
+					"command"),
+				"state_topic": hass.mqtt.getTopic(
+					"devices",
+					device.Dsid,
+					device.Name,
+					device.OutputChannels[0],
+					"state"),
+				"payload_on":  "100.00",
+				"payload_off": "0.00",
+				"qos":         0,
+			}
+			if device.Properties.Dimmable {
+				message["on_command_type"] = "brightness"
+				message["brightness_scale"] = 100
+				message["brightness_state_topic"] = hass.mqtt.getTopic(
+					"devices",
+					device.Dsid,
+					device.Name,
+					device.OutputChannels[0],
+					"state")
+				message["brightness_command_topic"] = hass.mqtt.getTopic(
+					"devices",
+					device.Dsid,
+					device.Name,
+					device.OutputChannels[0],
+					"command")
+
+			}
 
 		}
 	} else if device.DeviceType == digitalstrom.Blind {

